lib/matplotlib/rcsetup.py

```diff
@@ -299,6 +301,12 @@ def validate_color(s):
 
     raise ValueError('%s does not look like a color arg%s' % (s, msg))
 
+def deprecate_axes_colorcycle(value):
+    warnings.warn("axes.color_cycle is deprecated.  Use axes.prop_cycle "
+                  "instead. Will be removed in 2.1.0")
+    return validate_colorlist(value)
```

Comment:
@OceanWolf, We don't decorate the validate_colorlist() function because it is a perfectly valid function on its own for validating a list of colors, and so it could be used elsewhere in contexts that have nothing to do with `color_cycle`. You see this design elsewhere for other deprecated params.  

As for setting the `prop_cycle` in `validate_colorlist()`, this is just a validator function. These validators are used by the RcSetup class to validate inputs for every entry. The validators cannot, and should not, be setting any params.